File: 04_Trees_and_Graphs/src/sameTree/Solution.java
public class Solution {
  public static boolean sameTree(TreeNode tree1, TreeNode tree2) {
    // both are empty
    if(tree1 == null && tree2 == null) {
      return true;
    }
    // both are non empty, compare their data and recurse
    else if(tree != null && tree2 != null) {
        return tree1.data == tree2.data &&
               sameTree(tree1.left, tree2.left) &&
               sameTree(tree1.right, tree2.right);
    }
    // one is empty, other is not
    else {
      return false;
    }
  }
}

File: 04_Trees_and_Graphs/sameTree/solution.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} p
 * @param {TreeNode} q
 * @return {boolean}
 */
var isSameTree = function(p, q) {
  if (!p || !q) {
      return p === q
  }
  return p.val === q.val && isSameTree(p.left, q.left) && isSameTree(p.right, q.right)
};